packages/babel-parser/src/parser/expression.js

```diff
@@ -2611,4 +2628,24 @@ export default class ExpressionParser extends LValParser {
 
     return ret;
   }
+
+  // https://github.com/tc39/proposal-js-module-blocks
+  parseModuleExpression(): N.ModuleExpression {
+    this.expectPlugin("jsModuleBlocks");
+    const node = this.startNode<N.ModuleExpression>();
+    node.body = [];
+    this.next(); // eat "module"
+    const oldLabels = this.state.labels;
+    this.state.labels = [];
+    const oldExportedIdentifiers = this.state.exportedIdentifiers;
+    this.state.exportedIdentifiers = [];
+    this.eat(tt.braceL);
+    this.scope.enter(SCOPE_OTHER);
+    this.parseBlockOrModuleBlockBody(node.body, undefined, true, tt.braceR);
```

Comment:
Can you a test case about the locality of `this.inModule`? We should throw for
```js
// sourceType: "script"
module { await: 1 }
```
because "await" as a label identifier is forbidden when the goal syntax is Module: https://tc39.es/ecma262/#sec-identifiers-static-semantics-early-errors

Reply:
b4adac3 👍